crates/http/src/routes.rs

```diff
@@ -47,37 +51,47 @@ impl Router {
         Ok((Self { routes }, duplicates))
     }
 
-    /// This returns the component ID that should handle the given path, or an error
-    /// if no component matches.
-    ///
-    /// If multiple components could potentially handle the same request based on their
-    /// defined routes, components with matching exact routes take precedence followed
-    /// by matching wildcard patterns with the longest matching prefix.
-    pub(crate) fn route(&self, p: &str) -> Result<&str> {
+    /// Returns the constructed routes.
+    pub fn routes(&self) -> &IndexMap<RoutePattern, String> {
```

Comment:
It doesn't look like this method is used yet; whats the plan for it? If we can avoid exposing the inner map type it would be easier to change.

Reply:
It's used by the `http-trigger` crate to print the available routes.

Reply:
https://github.com/fermyon/spin/blob/12a038a3b29039dc31dc0a2d6752a8059a414d04/crates/http-trigger/src/lib.rs#L182

Reply:
Cool; could it be `-> impl Iterable<Item = (&RoutePattern, &String)>` or whatever `IndexMap::iter` returns?

Reply:
Ah I missed the "Large diffs are not rendered by default." among all the "This file was deleted." :sweat_smile: